### Теория вероятностей и математическая статистика (семинары).
#### Урок 6. Сравнение долей. Построение доверительного интервала.

#### Задача 1.  
Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.  
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,  
если выборочная средняя M = 80, а объем выборки n = 256.

#### Решение:   
$\gamma=0,95$  
$\bar x=80$ среднее по выборке  
$\sigma=16$ среднее квадратическое отклонение  
$n=256$ обьем выборки  
Найдем доверительный интервал для математического ожидания
надежностью 0,95, по формуле:  
$\bar x-t_\gamma\frac{\sigma}{\sqrt{n}}<\alpha<\bar x+t_\gamma\frac{\sigma}{\sqrt{n}},$  
где $t_\gamma$ определим из таблицы по условию $Ф(t_\gamma)=\frac{\gamma}{2}=\frac{0,95}{2}=0,475=>t_\gamma\approx1,96$  
Получаем после подстановки известных данных:  
$80-1,96\frac{16}{\sqrt{256}}<\alpha<80+1,96\frac{16}{\sqrt{256}}$ 

In [2]:
left=80-1.96*16/256**(1/2)
right=80+1.96*16/256**(1/2)
print(f'95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [{left};{right}].')

95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [78.04;81.96].


#### Ответ:  
Доверительный интервал для оценки математического ожидания a с надежностью 0.95 лежит в пределах [78,04 ; 81,96].

#### Задача 2.  
В результате 10 независимых измерений некоторой величины X, выполненных с
одинаковой точностью, получены опытные данные:  
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1  
Предполагая, что результаты измерений подчинены нормальному закону распределения
вероятностей,  
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.  
#### Решение:  
Из условия задачи  не известно ни о мат. ожидании генеральной совокупности, ни о среднем квадратическом отклонении для генеральной совокупности, то для расчета $95\%$-го доверительного интервала будем использовать $t$-критерий, и формулу:  
$\bar X\pm t_{\alpha/2}\cdot\frac{\sigma_{несмещ}}{\sqrt{n}}$, где  
$\sigma_{несмещ}$-несмещённое среднее квадратическое отклонение выборки, $n$-размер выборки  
Используем библиотеку numpy:

In [7]:
import numpy as np

arr=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
print(f'Среднее выборочное: {np.mean(arr): .2f},\n'
      f'Размер выборки n={len(arr)},\n'
      f'Среднее квадратическое отклонение по выборке(несмещенное): {np.std(arr, ddof=1): .2f}.'
     )

Среднее выборочное:  6.59,
Размер выборки n=10,
Среднее квадратическое отклонение по выборке(несмещенное):  0.45.


Для вычисления $t$-критерия воспользуемся библиотекой scipy:

In [8]:
import scipy.stats as stats

def t_from_table(confidens, len_array):
    alpha=(1-confidens)
    return stats.t.ppf(1-alpha/2, len_array-1)
print(f'Табличное значение t-критерия для 95%-го доверительного интервала данной выборки: {t_from_table(0.95, len(arr)): .3f}')

Табличное значение t-критерия для 95%-го доверительного интервала данной выборки:  2.262


Функция для расчета доверительного интервала:

In [13]:
def confidens_int(arr, confidens):
    return round(np.mean(arr)-t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3), \
           round(np.mean(arr)+t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3)

print(f'95%-й доверительный интервал для истинного значения Х:{confidens_int(arr, 0.95)}.')

95%-й доверительный интервал для истинного значения Х:(6.268, 6.912).


#### Ответ:  
95%-й доверительный интервал для истинного значения Х : (6.268, 6.912).

#### Задача 3.  
Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170  
Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175  
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.
#### Решение:  
Найдем разность среднего $Sdelta$, воспользуемся библиотекой numpy:  

In [3]:
import math
import numpy as np
import scipy.stats as stats

array1 = [175, 167, 154, 174, 178, 148, 160, 167, 169, 170]
array2 = [178, 165, 165, 173, 168, 155, 160, 164, 178, 175]
n = len(array1)
a = 0.05
m1 = np.mean(array1)
m2 = np.mean(array2)
delta = abs(m1 - m2)
print(f'Разность средннего Sdelta = {delta: .4f}')

Разность средннего Sdelta =  1.9000


Найдем табличное значение $t$-критерия, для $\alpha=0.05$

In [7]:
t = stats.t.ppf(1 - (a / 2), 2 * (n - 1))
print(f'Табличное значение t-критерия = {t: .4f}')

Табличное значение t-критерия =  2.1009


Найдем дисперсию двух групп:

In [8]:
d = (np.var(array1, ddof=1) + np.var(array2, ddof=1)) / 2
print(f'Дисперсия двух групп = {d: .4f}')

Дисперсия двух групп =  75.5833


Находим $Sdelta$ и доверительный интервал доверительной вероятностью 0,95:

In [28]:
s = math.sqrt((d / n) + (d / n))
print(f'Sdelta = {s: .4f}')
result1 = delta - t * s
result2 = delta + t * s
print(
    f'Доверительный интервал для разности среднего роста родителей и детей с доверительной вероятностью 0,95 = [{result1: .4f};{result2: .4f}]')


Разность средннего =  1.9000
Табличное значение t-критерия =  2.1009
Дисперсия двух групп =  75.5833
Sdelta =  3.8880
Доверительный интервал для разности среднего роста родителей и детей с доверительной вероятностью 0,95 = [-6.2684; 10.0684]


#### Ответ:  
Доверительный интервал для разности среднего роста родителей и детей с доверительной вероятностью 0,95 = [-6.2684; 10.0684]